qt-bugs@ issue : N218280
Trolltech task ID : 218280
bugs.kde.org number : none
applied: yes
author: TT

QMainWindow::restoreState doesn't layout toolbars properly, regression
from Qt-4.4

--- a/src/gui/widgets/qtoolbararealayout.cpp
+++ b/src/gui/widgets/qtoolbararealayout.cpp
@@ -1323,7 +1323,8 @@
                 toolBar->setVisible(shown);
                 toolBar->d_func()->setWindowState(floating, true, rect);
 
-                item.extraSpace = item.size - pick(line.o, item.sizeHint());
+                //if it is -1, it means we should use the default size
+                item.extraSpace = (item.size == -1) ? 0 : item.size - pick(line.o, item.sizeHint());
 
 
                 line.toolBarItems.append(item);
